internal/testrunner/runners/system/servicedeployer/compose.go

```diff
@@ -76,47 +70,10 @@ func (r *DockerComposeServiceDeployer) SetUp(inCtxt ServiceContext) (DeployedSer
 	serviceContainer := fmt.Sprintf("%s_%s_1", service.project, serviceName)
 	outCtxt.Hostname = serviceContainer
 
-	// Redirect service container's STDOUT and STDERR streams to files in local logs folder
-	localLogsFolder := inCtxt.Logs.Folder.Local
-	agentLogsFolder := inCtxt.Logs.Folder.Agent
-
-	service.stdoutFilePath = filepath.Join(localLogsFolder, stdoutFileName)
-	logger.Debugf("creating temp file %s to hold service container %s STDOUT", service.stdoutFilePath, serviceContainer)
-	outFile, err := os.Create(service.stdoutFilePath)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not create STDOUT file")
-	}
-	service.stdout = outFile
```

Comment:
This is how a Docker-based service's STDOUT stream was being written to a log file on the local filesystem. How are we capturing a Docker-based service's logs now? 

Related: what would be the syntax in a package's test configuration file to refer to such logs? In the current implementation the syntax is `{{ STDOUT }}` and `{{ STDERR }}`.

Reply:
I considered this as a base assumption for this change - to base only on log file resources, as most apps emit them as two log streams (application, access). I think it simplified the implementation as we don't have to deal with opening and closing file.

In case of stdout, stderr we can "tee" these streams to local files.

I don't insist on such design, just wanted to handle all streams the same way (via filesystem).

Reply:
Okay, let me try to understand this with the `apache` Docker container. If you look at https://github.com/elastic/integrations/pull/263, the system test configuration file for the `apache/access` log dataset (for the current implementation of the system test runner) looks like this:

https://github.com/ycombinator/integrations/blob/124ce5912eaf010352c7054bbc0c5ec3e84664c4/packages/apache/dataset/access/_dev/test/system/config.yml

How do you see this changing in the new implementation (this PR)?

Reply:
We know in advance that service logs are available in the directory `/tmp/service_logs/` in the agent container. 

I suppose the variable `var.paths` can contain `/tmp/service_logs/access.log*`?

Reply:
Fixed. Additional context properties  are exposed: `LogsFolderLocal` and `LogsFolderAgent`.

Reply:
Thanks, so my understanding is that the system test configuration file for the `apache/access` log dataset would look like:

```
vars: ~
dataset:
  vars:
    paths:
      - "{{LogsFolderAgent}}/access.log*"
```

And `httpd.conf` would be configured as you showed in your last comment above.

But why would a package author just know that `logs/access_log` from their `httpd.conf` should be referred to as `{{LogsFolderAgent}}/access_log*`? I'm guessing this comes from the volume mount definitions that they would define in the docker-compose file for their service? Perhaps this last piece would be clear to me if you can make a PR to https://github.com/elastic/integrations/pull/263 showing the necessary changes.

Reply:
> Thanks, so my understanding is that the system test configuration file for the apache/access log dataset would look like:

Correct.

> And httpd.conf would be configured as you showed in your last comment above.

Yes

> But why would a package author just know that `logs/access_log` from their `httpd.conf` should be referred to as `{{LogsFolderAgent}}/access_log*`?

Frankly speaking I considered this as part of standard stack setup - the Elastic Agent has a directory bound to service logs of the service under tests (SUT). As the agent is a long running instance and it's booted up first, it needs to create the volume (no the SUT). The local directory with logs can customized on the SUT side:

```
version: '2.3'

services:
  apache:
    image: docker.elastic.co/integrations-ci/beats-apache:${SERVICE_VERSION:-2.4.20}-1
    ports:
      - 80
    volumes:
      - ${ELASTIC_PACKAGE_SERVICE_LOGS_DIR}:/usr/local/apache2/logs
``` 

To be honest I had in mind to document both `{{LogsFolderAgent}}` and `${ELASTIC_PACKAGE_SERVICE_LOGS_DIR}` in some README or guide.

I'm willing to adjust it if you have an idea on how we can simplify it.

Reply:
Okay, I understand the complete picture now. Thanks.

It is slightly annoying that these two variables expose the inner workings of how logs get from a service container to the agent container. It would be nice if a user didn't need to know about that detail. What if we make both variables (from a user's perspective) the same and something more generic, e.g. `SERVICE_LOGS_DIR`? Then, as a user, it is more intuitive that I'm mapping my service logs folder (`/usr/local/apache2/logs`) to `${SERVICE_LOGS_DIR}` and then in my system test config, reading from the same folder.

Reply:
Good call, it's definitely a better user experience.

Reply:
Fixed